packages/jest-haste-map/src/__tests__/fastpath-test.js

```diff
@@ -0,0 +1,575 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails oncall+jsinfra
+ *
+ * original author: dead_horse <dead_horse@qq.com>
+ * ported by: yaycmyk <evan@yaycmyk.com>
+ */
+
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* eslint-disable fb-www/soft-max-len */
+'use strict';
+
+jest.unmock('../fastpath');
+
+let fp;
+
+describe('fast-path', () => {
+
+  const actualPlatform = process.platform;
+  const invalidInputTests = [
+    true,
+    false,
+    7,
+    null,
+    {},
+    undefined,
+    [],
+    NaN,
+    () => {},
+  ];
+
+  const getErrorMessage = (fnName, test, expected, actual) => {
+    return [
+      `fp.${fnName}('${test}')`,
+      `expect=${JSON.stringify(expected)}`,
+      `actual=${JSON.stringify(actual)}`,
+    ].join('\n');
+  };
+
+  beforeEach(() => {
+    process.platform = 'linux';
+  });
+
+  afterEach(() => {
+    process.platform = actualPlatform;
+  });
+
+  describe('basename', () => {
+    it('should resolve given paths', () => {
+      process.platform = actualPlatform;
+
+      fp = require('../fastpath');
```

Comment:
I can't because we modify `process.platform` in each test and the module depends on it in the module factory. We wipe it away on every test, change the platform and freshly require fastpath.

The reason why this works differently in the node-haste repo right now is because we use the inline-requires transform there. This repo doesn't have a build step, so I'm just doing the requires manually :)